sopel/modules/weather.py

```diff
@@ -161,18 +159,17 @@ def say_info(bot, trigger, mode):
         if woeid is None:
             first_result = woeid_search(location)
             if first_result is not None:
-                woeid = first_result.get('woeid')
+                woeid = first_result
 
     if not woeid:
         return bot.reply("I don't know where that is.")
 
-    query = 'q=select * from weather.forecast where woeid="%s" and u=\'c\'' % woeid
-    body = requests.get('https://query.yahooapis.com/v1/public/yql?' + query)
-    parsed = xmltodict.parse(body.text).get('query')
-    results = parsed.get('results')
+    query = woeid
+    body = requests.get('https://www.metaweather.com/api/location/' + str(query))
```

Comment:
I think you should need to encode the `query` as an URL first. I might be wrong, but that seem unsafe as is.